py/farm_ng/track/utils.py

```diff
@@ -45,3 +54,314 @@ def update_filter_track(track_path: Path) -> None:
+            # Calculate the pose for the current segment
+            turn_segment: Pose3F64 = Pose3F64(
+                a_from_b=Isometry3F64([delta_distance, 0, 0], Rotation3F64.Rz(delta_angle)),
+                frame_a=segment_poses[-1].frame_b,
+                frame_b=f"{next_frame_b}_{i - 1}",
+            )
+            segment_poses.append(segment_poses[-1] * turn_segment)
+
+        # Rename the last pose to the desired name
+        segment_poses[-1].frame_b = next_frame_b
+        self.track_waypoints.extend(segment_poses)
+        self._segment_indices.append(len(self.track_waypoints))
+        self._loaded = False
+
+    def pop_last_segment(self) -> None:
+        """Remove the last (appended) segment from the track."""
+
+        if self._loaded:
+            print("Cannot pop segment from a loaded track without inserting new segments first.")
+            return
+
+        if len(self._segment_indices) > 1:  # Ensure there is a segment to pop
+            last_segment_start = self._segment_indices[-2]  # Get the start of the last segment
+            # Remove the waypoints from the last segment
+            self.track_waypoints = self.track_waypoints[:last_segment_start]
+            # Remove the last segment index
+            self._segment_indices.pop()
+        else:
+            print("No segment to pop.")
+
+    def unpack_track(self) -> tuple[list[float], list[float], list[float]]:
+        """Unpack x and y coordinates and heading from the waypoints for plotting.
+
+        Args: None
+        Returns:
+            tuple[list[float], list[float], list[float]]: The x, y, and heading coordinates of the track waypoints.
+        """
+
+        x: list[float] = []
+        y: list[float] = []
+        heading: list[float] = []
+        for pose in self.track_waypoints:
+            x.append(pose.a_from_b.translation[0])
+            y.append(pose.a_from_b.translation[1])
+            heading.append(pose.a_from_b.rotation.log()[-1])
+        return (x, y, heading)
+
+    def save_track(self, path: Path) -> None:
+        """Save the track to a json file.
+
+        Args:
+            path (Path): The path of the file to save.
+        """
+        if self.track:
+            proto_to_json_file(path, self.track)
+            print(f"Track saved to {path}")
+        else:
+            print("No track to save.")
+
+    def load_track(self, path: Path) -> None:
+        """Import a track from a json file.
+
+        Args:
+            path (Path): The path of the file to import.
+        """
+        loaded_track = proto_from_json_file(path, Track())
+        self.track = loaded_track
+
+    def merge_tracks(self, track_to_merge: Track, threshold: float = 0.5) -> bool:
+        """Merge a track with the current track.
+
+        Args:
+            track (Track): The track to merge.
+        """
+        # TODO: Check if we can invert the track_to_merge and append it to the current track
```

Comment:
Not entirely sure what this means, but consider addressing the TODO before merge

```suggestion
        # TODO: Check if we can invert the track_to_merge and append it to the current track
```